src/App/Action/Backend/Order/PostOrderNotesAction.php

```diff
@@ -52,12 +54,22 @@ public function handle(Request $request): Response
                 return $order->apiIdentifier && $order->notes->isNotEmpty();
             })
             ->each(function (PdkOrder $order) {
-                $this->orderNotesRepository->postOrderNotes(
+                $notes = $this->orderNotesRepository->postOrderNotes(
                     $order->apiIdentifier,
-                    $order->notes->toFulfilmentCollection()
+                    $order->notes->filter(function (PdkOrderNote $note) {
+                        return null === $note->apiIdentifier;
+                    })
+                        ->toFulfilmentCollection()
                 );
+                $order->notes->each(function (PdkOrderNote $note) use ($notes) {
```

Comment:
Dit liever in een method op de PdkOrderNoteCollection, net als `addIds` en dergelijke dingen bij shipments en orders. Kijk dan ook hoe die zijn geïmplementeerd, want dat kan wat strakker :)

Reply:
plus, hoe "save" je dit nu? dit wordt nu niet opgeslagen in het platform (zie `PdkOrder::toStorableArray()`)